### DISCUSSION TITLE
Container WalkingAnimationMapping

### DISCUSSION THREAD

**Discussion by lendsp** [Annoraaq/grid-engine#174] [Ideas]
Hi, I was thinking about the possibility of players equipping clothes in my game. and to not have the trouble of making all possible sprites, it would be interesting to grid configure this when passed in the container.

![BF101100100](https://user-images.githubusercontent.com/21299848/126009565-53a1e903-1035-4508-b4d6-7b7deae21cad.png)
![ropa](https://user-images.githubusercontent.com/21299848/126009591-ae56fb04-1391-4401-9da0-8ce77365a243.png)

http://34.95.166.209/teste.html

If you want to see how my game is doing, the link is: http://34.95.166.209/index.html

**Comment by Annoraaq** [comment]
I think when we provide a method 'setSprite' for characters it should be easy to replace the sprite.

Until then I fear that you will either have to replace the whole character or create a single sprite with all the clothes and create a custom animation.

**Comment by lendsp** [comment]
i found a way, sorry about draft, 
I know it's not difficult to do some variables to calculate, I suffered a bit to assimilate the second parameter of texture.add.
the exchange on the grid, I can do it with setposition/remove, but it will be better when I have setsprite

the result was very good :D
http://34.95.166.209/teste.html

const texture = this.textures.createCanvas('teste', 128, 196);
this.textures.get('body').getSourceImage();
this.textures.get('ropa').getSourceImage();
//
texture.drawFrame('body', 0, 0, 0);
texture.drawFrame('ropa', 0, 0, 0);
texture.drawFrame('body', 1, 32, 0);
texture.drawFrame('ropa', 1, 32, 0);
texture.drawFrame('body', 2, 64, 0);
texture.drawFrame('ropa', 2, 64, 0);
texture.drawFrame('body', 3, 96, 0);
texture.drawFrame('ropa', 3, 96, 0);
//
texture.drawFrame('body', 4, 0, 48);
texture.drawFrame('ropa', 4, 0, 48);
texture.drawFrame('body', 5, 32, 48);
texture.drawFrame('ropa', 5, 32, 48);
texture.drawFrame('body', 6, 64, 48);
texture.drawFrame('ropa', 6, 64, 48);
texture.drawFrame('body', 7, 96, 48);
texture.drawFrame('ropa', 7, 96, 48);
//
texture.drawFrame('body', 8, 0, 96);
texture.drawFrame('ropa', 8, 0, 96);
texture.drawFrame('body', 9, 32, 96);
texture.drawFrame('ropa', 9, 32, 96);
texture.drawFrame('body', 10, 64, 96);
texture.drawFrame('ropa', 10, 64, 96);
texture.drawFrame('body', 11, 96, 96);
texture.drawFrame('ropa', 11, 96, 96);
//
texture.drawFrame('body', 12, 0, 144);
texture.drawFrame('ropa', 12, 0, 144);
texture.drawFrame('body', 13, 32, 144);
texture.drawFrame('ropa', 13, 32, 144);
texture.drawFrame('body', 14, 64, 144);
texture.drawFrame('ropa', 14, 64, 144);
texture.drawFrame('body', 15, 96, 144);
texture.drawFrame('ropa', 15, 96, 144);
//
texture.add(0,0, 0, 0, 32, 48);
texture.add(1,0, 32, 0, 32, 48);
texture.add(2,0, 64, 0, 32, 48);
texture.add(3,0, 96, 0, 32, 48);
//
texture.add(4,0, 0, 48, 32, 48);
texture.add(5,0, 32, 48, 32, 48);
texture.add(6,0, 64, 48, 32, 48);
texture.add(7,0, 96, 48, 32, 48);
//
texture.add(8,0, 0, 96, 32, 48);
texture.add(9,0, 32, 96, 32, 48);
texture.add(10,0, 64, 96, 32, 48);
texture.add(11,0, 96, 96, 32, 48);
//
texture.add(12,0, 0, 144, 32, 48);
texture.add(13,0, 32, 144, 32, 48);
texture.add(14,0, 64, 144, 32, 48);
texture.add(15,0, 96, 144, 32, 48);
//
player = this.add.sprite(0,0,'teste').setScale(2);
//
self.gridEngine.addCharacter({id: 'player', sprite: player , walkingAnimationMapping: wm, startPosition: {x: 11, y: 6 }, speed: 3});

**Comment by Annoraaq** [comment]
The setSprite method is available in the freshly released version 2.6.0 now. I hope this solves your problem.
